src/charm.py

```diff
@@ -401,6 +408,62 @@ def _is_magmad_enabled(self) -> bool:
         )
         return not magma_service.returncode
 
+    def _install_configurations(self, event: OrchestratorAvailableEvent) -> bool:
+        """Install or update configuration files.
+
+        Returns:
+            True if any changes were applied
+        """
+        config = self._generate_config(event)
+        return any(
+            [
+                self._install_file(Path(ROOT_CA_PATH), event.root_ca_certificate),
+                self._install_file(Path(CONFIG_PATH), config),
+            ]
+        )
+
+    @staticmethod
+    def _install_file(file: Path, content: str) -> bool:
+        """Install file with provided text content.
+
+        Args:
+            file: Path object to write to
+            content: Text content to write to the file
+
+        Returns:
+            True if the file was written to
+        """
+        if not file.parent.exists():
+            file.parent.mkdir()
+        elif file.exists() and file.read_text() == content:
+            return False
+        file.write_text(content)
+        return True
+
+    @staticmethod
+    def _generate_config(event: OrchestratorAvailableEvent) -> str:
```

Comment:
It feels like this method should take addresses and ports as inputs instead of the event object. This way we could generate the same config file based on other triggers than the relation event.

Reply:
Not that we will do that btw, it's just that I don't see why this method would **need** the event object.

Reply:
Fixed